sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ReceiveMessageWithTransactionAsyncSample.java

```diff
@@ -0,0 +1,108 @@
+        String connectionString = "Endpoint={fully-qualified-namespace};SharedAccessKeyName={policy-name};"
+            + "SharedAccessKey={key}";
+
+        // Create a receiver.
+        // "<<fully-qualified-namespace>>" will look similar to "{your-namespace}.servicebus.windows.net"
+        // "<<queue-name>>" will be the name of the Service Bus queue instance you created
+        // inside the Service Bus namespace.
+        ServiceBusClientBuilder builder = new ServiceBusClientBuilder()
+            .connectionString(connectionString);
+
+        // At most, the receiver will automatically renew the message lock until 120 seconds have elapsed.
+        // By default, after messages are processed, the onus is on users to complete, abandon, defer, or dead-letter the
+        // message when they are finished with it.
+
+        ServiceBusReceiverAsyncClient receiver = builder
+            .receiver()
+            .receiveMode(ReceiveMode.PEEK_LOCK)
+            .queueName("<<queue-name>>")
+            .maxAutoLockRenewalDuration(Duration.ofSeconds(120))
+            .buildAsyncClient();
+
+        ServiceBusSenderAsyncClient sender = builder.sender()
+            .queueName("<<queue-name>>")
+            .buildAsyncClient();
+
+        List<ServiceBusMessage> messages = new ArrayList<>();
+        messages.add(new ServiceBusMessage("Hello world!".getBytes(UTF_8)));
+        messages.add(new ServiceBusMessage("Hello world!".getBytes(UTF_8)));
+
+        // Send some messages.
+        sender.send(messages).block();
+
+        // Hold the transaction.
+        AtomicReference<ServiceBusTransactionContext> transaction = new AtomicReference<>();
+
+        // Transaction is actually started in ServiceBus until you perform first operation
+        // (Example receiver.complete(message, transaction)) with it.
+        // Create transaction and start receiving messages
+        Disposable subscription = receiver.createTransaction()
+            .flatMapMany(transactionContext -> {
+                transaction.set(transactionContext);
+                return receiver.receive();
+            })
+            .take(2)
+            .flatMap(context -> {
+                boolean messageProcessed = true;
+                // Process the context and its message here.
+                // Change the `messageProcessed` according to you business logic and if you are able to process the
+                // message successfully.
+
+                if (messageProcessed) {
+                    return receiver.complete(context.getMessage(), transaction.get());
+                } else {
+                    return receiver.abandon(context.getMessage(), null, transaction.get());
+                }
+            })
+            .doFinally(signal -> {
+                if (signal == SignalType.ON_COMPLETE) {
+                    receiver.commitTransaction(transaction.get()).subscribe();
```

Comment:
This callback/method could end long before this completes and the receiver is disposed.